statsmodels/distributions/joint_distribution.py

```diff
@@ -0,0 +1,236 @@
+    transform of a distribution function if and only if :math:`\phi` is
+    completely monotone and :math:`\phi(0) = 1` [2]_.
+
+    The following algorithm for sampling a ``d``-dimensional exchangeable
+    Archimedean copula with generator :math:`\phi` is due to Marshall, Olkin
+    (1988) [1]_, where :math:`LS^{−1}(\phi)` denotes the inverse
+    Laplace-Stieltjes transform of :math:`\phi`.
+
+    From a mixture representation with respect to :math:`F`, the following
+    algorithm may be derived for sampling Archimedean copulas, see [1]_.
+
+    1. Sample :math:`V \sim F = LS^{−1}(\phi)`.
+    2. Sample i.i.d. :math:`X_i \sim U[0,1], i \in \{1,...,d\}`.
+    3. Return:math:`(U_1,..., U_d)`, where :math:`U_i = \phi(−\log(X_i)/V), i
+       \in \{1, ...,d\}`.
+
+    Detailed properties of each copula can be found in [3]_.
+
+    Instances of the class can access the attributes: ``rng`` for the random
+    number generator (used for the ``seed``).
+
+    **Subclassing**
+
+    When subclassing `Copula` to create a new copula, ``__init__`` and
+    ``random`` must be redefined.
+
+    * ``__init__(seed=None)``: If the sampler
+      does not take advantage of a ``seed``, this parameter can be omitted.
+    * ``random(n)``: draw ``n`` from the copula.
+
+    References
+    ----------
+    .. [1] Marshall AW, Olkin I. “Families of Multivariate Distributions”,
+      Journal of the American Statistical Association, 83, 834–841, 1988.
+    .. [2] Marius Hofert. "Sampling Archimedean copulas", Universität Ulm, 2008.
+    .. [3] Harry Joe. "Dependence Modeling with Copulas", Monographs on
+      Statistics and Applied Probability 134, 2015.
+
+    """
+
+    @abstractmethod
+    def __init__(self, seed=None):
+        self.rng = np.random.default_rng(seed)
+
+    @abstractmethod
+    def random(self, n):
+        """Draw `n` in the half-open interval ``[0, 1)``.
+
+        Marginals are uniformly distributed.
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate in the parameter space.
+            Default is 1.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            Sample from the copula.
+
+        """
+
+
+class IndependentCopula(Copula):
+    """Independent copula."""
+
+    def __init__(self, seed=None):
+        super().__init__(seed=seed)
+
+    def random(self, n):
+        x = self.rng.random((n, 2))
+        # v = ...
+        return x  # np.exp(- (-np.log(x) / v))
+
+
+
+class GaussianCopula(Copula):
+    """Gaussian copula."""
+
+    def __init__(self, cov=1, seed=None):
+        super().__init__(seed=seed)
+        self.density = stats.norm()
+        self.mv_density = stats.multivariate_normal(cov=cov)
+
+    def random(self, n):
+        x = self.mv_density.rvs(size=n, random_state=self.rng)
+        return self.density.cdf(x)
+
+
+class StudentCopula(Copula):
+    """Student copula."""
```

Comment:
docstrings should have a reasonable amount of mathematicaal detail similar to what scipy does.

Reply:
ok I will see to add at least the definition of C. But I am not a mathematician and I would just copy paste. So if you know better, it's better if you do it.

Reply:
> ok I will see to add at least the definition of C. But I am not a mathematician and I would just copy paste. So if you know better, it's better if you do it.

Copy-paste from Wiki is a good start.